lib/Twig/Environment.php

```diff
@@ -313,6 +315,10 @@ public function getTemplateClass($name, $index = null)
         $key .= json_encode(array_keys($this->extensions));
         $key .= function_exists('twig_template_get_attributes');
         $key .= ':'.PHP_MAJOR_VERSION.':'.PHP_MINOR_VERSION;
+        $key .= ':debug:'.($this->debug ? '1' : '0');
+        $key .= ':charset:'.$this->charset;
+        $key .= ':base_template_class:'.$this->baseTemplateClass;
+        $key .= ':strict_variables:'.($this->strictVariables ? '1' : '0');
```

Comment:
Instead of computing this all the time, what about storing the value in the class constructor and just reusing it here? What about adding Twig's version?